### PR TITLE
#45: Added initial section on supply chain attacks.

### PR DESCRIPTION
**Pull request by sam-ellis**

### CODE REVIEW

File: book.md [hunk 1]
```diff
@@ -251,6 +251,70 @@ also need to keep cache side-channel attacks in mind, which are discussed in the
 for suggestions of what this should contain.}
 <!-- markdown-link-check-enable-->
 
+# Supply chain attacks
+
+A software _supply chain attack_ occurs when an attacker interferes with the
+software development or distribution processes to impact end-users of that
+software. Supply chain attacks and their possible mitigations are not specific
```
[the remark this answers is not in the document]
Comment (sam-ellis) [reply book.md:258]: Accepting suggested modification to text.

File: book.md [hunk 2]
```diff
@@ -251,6 +251,70 @@ also need to keep cache side-channel attacks in mind, which are discussed in the
...
+of the ubiquity of compiler technology, to developers, in continuous
+integration systems and as JITs, and because such attacks can go on to silently
+infect many other applications that were created with or run using those
+compilers. For this reason it is worth exploring the history of attacks that
+involve compilers and what can be done to prevent them.
+
+As far back as 1974 Karger & Schell theorized about an attack on the Multics
+operating system via the PL/I compiler [@Karger1974]:
+
+> In Multics, most of the ring 0 supervisor is written in PL/1. A penetrator
```
[the remark this answers is not in the document]
Comment (sam-ellis) [reply book.md:269]: Accepted suggestion to remove quote.

File: book.md [hunk 3]
```diff
@@ -251,6 +251,70 @@ also need to keep cache side-channel attacks in mind, which are discussed in the
...
+If these cases seem far-fetched then consider that there have been several real
+examples of supply chain attacks on development tools.
+
+Induc is family of viruses that infects a pre-compiled library in the Delphi
+toolchain with malicious code [@Gostev2009]. When Delphi compiles a project the
+malicious library is included into the resulting executable, thus enabling the
+virus to spread. The virus was first detected in 2009 and was circulating
+undetected for at least a year beforehand. The Induc-B variant infects two other
+toolchains in addition to Delphi. Thousands of infections were reported, and
+some in commonly used applications.
```
[the remark this answers is not in the document]
Comment (sam-ellis) [reply book.md:294]: The data that is readily available from one anti-virus vendor shows monthly **infected computers** as follows:

- Oct 2009: 49k (https://securelist.com/monthly-malware-statistics-october-2009/36271/)
- Nov 2009: 46k (https://securelist.com/monthly-malware-statistics-november-2009/36276/)
- Dev 2009: 37k (https://securelist.com/monthly-malware-statistics-december-2009/36281/)

These total > 100k **infected computers**. This is of course data from only one vendor for three months only. So true infection rate is likely to be higher, but don't early have the data available to confirm this.

Based on this, I am clarifying the text to read "in excess of a hundred thousand infected computers were detected world-wide by anti-virus solutions".

[the remark this answers is not in the document]
Comment (kbeyls) [reply book.md:294]: Great research, thank you!

**Review by kbeyls — APPROVED** [review]
Thanks Sam, I think this looks good to commit/merge.
I only have 2 nitpicky suggestions for what I think is a missing word:
- "tracked down a trojanized version" -> "tracked down to a trojanized version"?
- "Induc is family of viruses" -> "Induc is a family of viruses"?

File: book.md [hunk 4]
```diff
@@ -251,6 +251,67 @@ also need to keep cache side-channel attacks in mind, which are discussed in the
...
+malicious library is included into the resulting executable, thus enabling the
+virus to spread. The virus was first detected in 2009 and was circulating
+undetected for at least a year beforehand. Several popular applications are
+known to have been infected, including a chat client and a media player.
+Overall, in excess of a hundred thousand infected computers were detected
+world-wide by anti-virus solutions.
+
+XcodeGhost is the name given to malware first detected in 2015 that infected
+thousands of iOS applications [@Cox2015]. The source of the infection was
+tracked down a trojanized version of Xcode tools. The malware exists
```
[the remark this answers is not in the document]
Comment (sam-ellis) [reply book.md]: Fixed in latest pull request update.

**Pull request merged**